egui/src/widgets/plotting/plot.rs

```diff
@@ -480,170 +272,86 @@ impl Widget for Plot {
-                transform.shift_bounds(-response.drag_delta());
-                auto_bounds = false;
-            }
+        // Zooming
+        if allow_zoom {
             if let Some(hover_pos) = response.hover_pos() {
                 let scroll_delta = ui.input().scroll_delta[1];
                 if scroll_delta != 0. {
                     transform.zoom(-0.01 * scroll_delta, hover_pos);
                     auto_bounds = false;
                 }
             }
-
-            ui.memory().id_data.insert(
-                plot_id,
-                PlotMemory {
-                    bounds: *transform.bounds(),
-                    auto_bounds,
-                },
-            );
-
-            let prepared = Prepared {
-                curves,
-                hlines,
-                vlines,
-                show_x,
-                show_y,
-                transform,
-            };
-            prepared.ui(ui, &response);
         }
 
-        response.on_hover_cursor(CursorIcon::Crosshair)
-    }
-}
-
-/// Contains the screen rectangle and the plot bounds and provides methods to transform them.
-struct ScreenTransform {
-    /// The screen rectangle.
-    rect: Rect,
-    /// The plot bounds.
-    bounds: Bounds,
-}
-
-impl ScreenTransform {
-    fn rect(&self) -> &Rect {
-        &self.rect
-    }
-
-    fn bounds(&self) -> &Bounds {
-        &self.bounds
-    }
-
-    fn shift_bounds(&mut self, mut delta_pos: Vec2) {
-        delta_pos.x *= self.dvalue_dpos()[0] as f32;
-        delta_pos.y *= self.dvalue_dpos()[1] as f32;
-        self.bounds.translate(delta_pos);
-    }
-
-    /// Zoom by a relative amount with the given screen position as center.
-    fn zoom(&mut self, delta: f32, center: Pos2) {
-        let delta = delta.clamp(-1., 1.);
-        let rect_width = self.rect.width();
-        let rect_height = self.rect.height();
-        let bounds_width = self.bounds.width() as f32;
-        let bounds_height = self.bounds.height() as f32;
-        let t_x = (center.x - self.rect.min[0]) / rect_width;
-        let t_y = (self.rect.max[1] - center.y) / rect_height;
-        self.bounds.min[0] -= ((t_x * delta) * bounds_width) as f64;
-        self.bounds.min[1] -= ((t_y * delta) * bounds_height) as f64;
-        self.bounds.max[0] += (((1. - t_x) * delta) * bounds_width) as f64;
-        self.bounds.max[1] += (((1. - t_y) * delta) * bounds_height) as f64;
-    }
-
-    fn position_from_value(&self, value: &Value) -> Pos2 {
-        let x = remap(
-            value.x,
-            self.bounds.min[0]..=self.bounds.max[0],
-            (self.rect.left() as f64)..=(self.rect.right() as f64),
-        );
-        let y = remap(
-            value.y,
-            self.bounds.min[1]..=self.bounds.max[1],
-            (self.rect.bottom() as f64)..=(self.rect.top() as f64), // negated y axis!
+        // Initialize values from functions.
+        // TODO: Let the user pick a resolution?
```

Comment:
The `from_explicit_callback` and `from_parametric_callback` already does, so this `TODO` is done?